### DISCUSSION TITLE
Improving Python Project Configuration

### DISCUSSION THREAD

**Discussion by downdawn** [fastapi-practices/fastapi_best_architecture#22] [Ideas]
## 1. Email related configuration is useless

All prefixed with 'EMAIL_'

## 2. Environment configuration

The original intention of `.env.example` is just an example.

`env_file = '.env'`

## 3. Log name

The name of FastBlog.log is not appropriate, and it is configured in conf
```
log_file = os.path.join(path_conf.LogPath, settings.LOG_NAME)
```
like
```
LOG_NAME: str = 'fba.log'
```

## 4. A friendlier "migration version"

`alembic.ini`

```
file_template = %%(year)d-%%(month).2d-%%(day).2d_%%(hour).2d-%%(minute).2d_%%(rev)s_%%(slug)s
```

## 5. Use python3.10 version

The pre-commit configuration specifies version 3.10 by default, and some new features can be used later, of course, compatibility should also be considered

[https://fastapi.tiangolo.com/python-types/#newer-versions-of-python](https://fastapi.tiangolo.com/python-types/#newer-versions-of-python)

**Comment by wu-clan** [comment]
Hi, @downdawn

I will directly PR 1,2,3,4

Create issue for 5

**Comment by wu-clan** [comment]
#23

**Comment by wu-clan** [comment]
#25

**Comment by wu-clan** [comment]
#28

**Comment by wu-clan** [comment]
#29
